src/sas/qtgui/Perspectives/Fitting/InViewWidget.py

```diff
@@ -0,0 +1,318 @@
+import copy
+import logging
+
+from PySide6 import QtCore, QtWidgets
+
+from sas.qtgui.Perspectives.Fitting import FittingUtilities
+from sas.qtgui.Perspectives.Fitting.ModelThread import Calc1D
+from sas.qtgui.Perspectives.Fitting.UI.InViewWidgetUI import Ui_InViewWidgetUI
+from sas.qtgui.Plotting.Plotter import PlotterWidget
+from sas.qtgui.Plotting.PlotterData import Data1D
+from sas.qtgui.Utilities import GuiUtils
+from sas.system import HELP_SYSTEM
+
+logger = logging.getLogger(__name__)
+
+class InViewWidget(QtWidgets.QWidget, Ui_InViewWidgetUI):
+    def __init__(self, parent: QtWidgets.QWidget):
+        super().__init__(parent)
+        self.setAttribute(QtCore.Qt.WA_DeleteOnClose, True)
+        self.setWindowFlag(QtCore.Qt.Window, True)
+        self.setupUi(self)
+        self.communicator = GuiUtils.Communicate()
+        self.plotter = PlotterWidget(self, self)
+        self._has_data = False
+        if self.plotBox.layout() is not None:
+            self.plotBox.layout().addWidget(self.plotter)
+        self._fw = None
+        self._plot_name = "InView Model"
+        self._param_values = {}
+        self._param_info = {}
+        self._slider_meta = {}
+        self._update_timer = QtCore.QTimer(self)
+        self._update_timer.setSingleShot(True)
+        self._update_timer.setInterval(75)
+        self._update_timer.timeout.connect(self._recompute_model)
+        if hasattr(self, 'cmdUpdateParam'):
+            self.cmdUpdateParam.clicked.connect(self._apply_to_fitpage)
+        if hasattr(self, 'cmdHelp'):
+            self.cmdHelp.clicked.connect(self.onHelp)
+        if hasattr(self, 'cmdClose'):
+            self.cmdClose.clicked.connect(self.close)
+
+    def setData(self, data):
+        if not isinstance(data, Data1D):
+            return
+        self._has_data = True
+        #self.plotter.plot(data=data)
+
+        try:
+            x = data.x
+            y = data.y
+            if y is not None and hasattr(y, "__len__") and len(x) == len(y):
+                self.plotter.plot(data=data)
+        except Exception:
+            pass
+
+    def initFromFitPage(self, fitting_widget):
+        self._fw = fitting_widget
+        params = list(self._fw.main_params_to_fit)
+        try:
+            if self._fw.chkPolydispersity.isChecked():
+                params += list(self._fw.polydispersity_widget.poly_params_to_fit)
+        except Exception:
+            pass
```

Comment:
The polydispersity and magnetic widget params to fit always exist and are always lists, so accessing them should never throw errors when accessing them and there should be no need to recast them to lists.

```
params = self._fw.main_params_to_fit

if self._fw.chkPolydispersity.isChecked():
            params += self._fw.polydispersity_widget.poly_params_to_fit
if self._fw.chkMagnetism.isChecked():
            params += self._fw.magnetism_widget.magnet_params_to_fit
```